packages/gatsby-plugin-offline/src/gatsby-node.js

```diff
@@ -1,19 +1,32 @@
 // use `let` to workaround https://github.com/jhnns/rewire/issues/144
 let fs = require(`fs`)
 let workboxBuild = require(`workbox-build`)
+const { promisify } = require(`util`)
 const path = require(`path`)
 const { slash } = require(`gatsby-core-utils`)
 const glob = require(`glob`)
 const _ = require(`lodash`)
 
+const readFile = promisify(fs.readFile)
+const writeFile = promisify(fs.writeFile)
 let getResourcesFromHTML = require(`./get-resources-from-html`)
 
-exports.createPages = ({ actions }) => {
+exports.onPreBootstrap = async ({ cache }) => {
+  const appShellDir = cache.directory
+  const appShellSourcePath = path.join(__dirname, `app-shell.js`)
+  const appShellTargetPath = path.join(appShellDir, `app-shell.js`)
+  const appShellSource = await readFile(appShellSourcePath)
+
+  await writeFile(appShellTargetPath, appShellSource)
```

Comment:
We can do `copyFile`? Also we can use sync methods because Gatsby runs APIs serially so there is nothing else going on (this will remove need for `promisify`)

Reply:
Updated :+1: